tools/autograd/derivatives.yaml

```diff
@@ -514,7 +514,7 @@
   result: self_t / other_p - other_t * (self_p / other_p) / other_p
 
 - name: div.Scalar(Tensor self, Scalar other) -> Tensor
-  self: div_tensor_self_backward(grad, at::scalar_to_tensor(other), self.scalar_type())
```

Comment:
If we can legitimately change these on master then it seems like a great fix.  @bdhirsh @ngimel any idea if this is safe or why it wasn't done this way before?

Reply:
I'm not sure using `at::native` calls in derivatives.yaml is allowed, @albanD do you know?

Reply:
It is never allowed to call into `at::native` if you're not already in `at::native` :)

Reply:
ok. We probably should have asked a broader question first; 

It seems like there is inconsistency in how scalars get wrapped in tensors.  Sometimes they have the 'wrapped number' flag, other times not.  Is there a reason for this, or is it better to make them consistent?  If we can set the wrapped_number flag here, but without using an at::native method, we'd be happy with that too obviously.

Reply:
> It seems like there is inconsistency in how scalars get wrapped in tensors.

Yes, I would even say that it is kind of a mess.
I don't think we have any strict rule on what should happen here. The only important point here is to make sure you don't introduce sync points with cuda.

Reply:
I think converting Scalar -> Tensor should always set wrapped_number flag, but it's also not a big deal if it doesn't. @mruberry was going to change type promotion to make this flag not matter. Why is setting wrapped number flag so important? It should not matter at all. Or do you have alternative implementation of wrapped_scalar_tensor that's different from scalar_to_tensor? In any case, go ahead with this change, just don't call `native`.

Reply:
Inside lazy tensor, we have a special case for wrapped numbers.  We allow them to be cpu tensors.  all other tensors we expect to already be lazy tensors, and we err if we get an operation with some lazy tensors and some cpu tensors.

We want to strictly allow wrapped numbers to be promoted automatically to lazy tensors, but, not sure if these can be differentiated from user-created tensors.

Can we just check tensor.dim()==0 and tensor.numel()==1 instead of checking is_wrapped_number?

Reply:
@desertfire has merged a temporarily fix for this kind of issues in #70075 to the staging branch. So, I'm going to remove the part of derivative.yaml here and produce a PR to fix the master instead.